python/Ganga/GPIDev/Schema/Schema.py

```diff
@@ -243,14 +248,22 @@ def _getDefaultValueInternal(self, attr, val=None, check=False):
                 defvalue = _found_attrs[stored_attr_key]
             else:
                 defvalue = config[attr]
+                from Ganga.GPIDev.Base.Proxy import isProxy
+                if isProxy(defvalue):
+                    raise GangaException("(1)Proxy found where it shouldn't be in the Config: %s" % stored_attr_key)
+                ## Just in case a developer puts the proxied object into the default value!
                 _found_attrs[stored_attr_key] = defvalue
         except (KeyError, Config.ConfigError):
             # hidden, protected and sequence values are not represented in config
             defvalue = item['defvalue']
+            from Ganga.GPIDev.Base.Proxy import isProxy
+            if isProxy(defvalue):
+                raise GangaException("(2)Proxy found where is shouldn't be in the Config" % stored_attr_key)
+            ## Just in case a developer puts the proxied object into the default value!
             _found_attrs[stored_attr_key] = defvalue
 
         # in the checking mode, use the provided value instead
-        if check:
+        if check is True:
```

Comment:
Yes but it's callable from external functions so I'd rather be implicit in-case we ever change our way of working with this class.